lampod-cli/src/main.rs

```diff
@@ -44,21 +48,61 @@ fn main() -> error::Result<()> {
     Ok(())
 }
 
+fn write_words_to_file<P: AsRef<Path>>(path: P, words: Option<String>) -> io::Result<()> {
+    let mut file = OpenOptions::new()
+        .write(true)
+        .create(true)
+        .truncate(true)
+        .open(path)?;
+
+    match words {
+        Some(ref value) => {
+            file.write_all(value.as_bytes())?;
+        }
+        None => {}
+    }
+
+    Ok(())
+}
+
+fn load_words_from_file<P: AsRef<Path>>(path: P) -> io::Result<Option<String>> {
+    let mut file = File::open(path)?;
+    let mut content = String::new();
+
+    file.read_to_string(&mut content)?;
+
+    if content.is_empty() {
+        Ok(None)
+    } else {
+        Ok(Some(content))
+    }
+}
+
 /// Return the root directory.
 fn run(args: LampoCliArgs) -> error::Result<()> {
-    let mnemonic = if args.restore_wallet {
-        let inputs: String = term::input(
-            "BIP 39 Mnemonic",
-            None,
-            Some("To restore the wallet, lampo needs a BIP39 mnemonic with words separated by spaces."),
-        )?;
-        Some(inputs)
+    let restore_wallet = args.restore_wallet;
+
+    // After this point the configuration is ready!
+    let mut lampo_conf: LampoConf = args.try_into()?;
+
+    let words_path = format!("{}/", lampo_conf.path());
+    let mnemonic = if restore_wallet {
+        if Path::new(&format!("{}/wallet.dat", words_path)).exists() {
+            // Load the mnemonic from the file
+            load_words_from_file(format!("{}/wallet.dat", words_path))?
+        } else {
+            // If file doesn't exist, ask for user input
+            let inputs: String = term::input(
+                "BIP 39 Mnemonic",
+                None,
+                Some("To restore the wallet, lampo needs the BIP39 mnemonic with words separated by spaces."),
+            )?;
+            Some(inputs)
+        }
     } else {
         None
     };
```

Comment:
How is this looking?

```suggestion
    let mnemonic = if restore_wallet {
        if Path::new(&format!("{}/wallet.dat", words_path)).exists() {
            // Load the mnemonic from the file
            load_words_from_file(format!("{}/wallet.dat", words_path))?
        } else {
            // If file doesn't exist, ask for user input
            let inputs: String = term::input(
                "BIP 39 Mnemonic",
                None,
                Some("To restore the wallet, lampo needs the BIP39 mnemonic with words separated by spaces."),
            )?;
            Some(inputs)
        }
    } else {
        if Path::new(&format!("{}/wallet.dat", words_path)).exists() {
            // if file exist and wallet is not restored, return error
            error::bail!("Wallet already exists. Please use --restore-wallet")
        } else {
            None
        }
    };
```

Reply:
Mh not sure can you explaing what you want to do with this if else code?

Reply:
The else case is used when lampod command is called without `--restore-wallet`. Here it checks if the wallet file exists or not. If it does it tells the user to use restore wallet. If it doesnt it will go ahead and mark the mnemonic as None.

Later this is checked and if mnemonic is none, new words are generated.

Reply:
This is better.

Reply:
What about https://github.com/vincenzopalazzo/lampo.rs/pull/295#issuecomment-2453061580 ?